src/phy.c

```diff
@@ -338,3 +338,164 @@ static int handle_ed_scan(struct nl802154_state *state,
+		if (tb_pan_desc[NL802154_ATTR_PAN_DESC_CHANNEL_NUM]) {
+		    printf("\tChannel Num  : %d\n", nla_get_u8(tb_pan_desc[NL802154_ATTR_PAN_DESC_CHANNEL_NUM]));
+		}
+		if (tb_pan_desc[NL802154_ATTR_PAN_DESC_CHANNEL_PAGE]) {
+		    printf("\tChannel Page : %d\n", nla_get_u8(tb_pan_desc[NL802154_ATTR_PAN_DESC_CHANNEL_PAGE]));
+		}
+		if (tb_pan_desc[NL802154_ATTR_PAN_DESC_SUPERFRAME_SPEC]) {
+		    printf("\tSF spec      : %x\n", nla_get_u8(tb_pan_desc[NL802154_ATTR_PAN_DESC_SUPERFRAME_SPEC]));
+		}
+		if (tb_pan_desc[NL802154_ATTR_PAN_DESC_GTS_PERMIT]) {
+			char *gts = nla_get_u32(tb_pan_desc[NL802154_ATTR_PAN_DESC_GTS_PERMIT]) ? "TRUE" : "FALSE";
+		    printf("\tGTS permit   : %s\n", gts);
+		}
+		if (tb_pan_desc[NL802154_ATTR_PAN_DESC_LQI]) {
+		    printf("\tLQI          : %x\n", nla_get_u8(tb_pan_desc[NL802154_ATTR_PAN_DESC_LQI]));
+		}
+		if (tb_pan_desc[NL802154_ATTR_PAN_DESC_TIME_STAMP]) {
+		    printf("\tTimestamp    : %x\n", nla_get_u32(tb_pan_desc[NL802154_ATTR_PAN_DESC_TIME_STAMP]));
+		}
+		if (tb_pan_desc[NL802154_ATTR_PAN_DESC_SEC_STATUS]) {
+		    printf("\tSec status   : %x\n", nla_get_u8(tb_pan_desc[NL802154_ATTR_PAN_DESC_SEC_STATUS]));
+		}
+		if (tb_pan_desc[NL802154_ATTR_PAN_DESC_SEC_LEVEL]) {
+		    printf("\tSec level    : %x\n", nla_get_u8(tb_pan_desc[NL802154_ATTR_PAN_DESC_SEC_LEVEL]));
+		}
+		if (tb_pan_desc[NL802154_ATTR_PAN_DESC_KEY_ID_MODE]) {
+		    printf("\tKey Id Mode  : %x\n", nla_get_u8(tb_pan_desc[NL802154_ATTR_PAN_DESC_KEY_ID_MODE]));
+		}
+		if (tb_pan_desc[NL802154_ATTR_PAN_DESC_KEY_SRC]) {
+		    printf("\tKey Src      : %x\n", nla_get_u8(tb_pan_desc[NL802154_ATTR_PAN_DESC_KEY_SRC]));
+		}
+		if (tb_pan_desc[NL802154_ATTR_PAN_DESC_KEY_INDEX]) {
+		    printf("\tKey Index    : %d\n", nla_get_u8(tb_pan_desc[NL802154_ATTR_PAN_DESC_KEY_INDEX]));
+		}
+	}
+
+	if (tb_msg[NL802154_ATTR_PEND_ADDR_SPEC]) {
+		printf("Pend addr spec: %x\n", nla_get_u8(tb_msg[NL802154_ATTR_PEND_ADDR_SPEC]));
+	}
+
+	if (tb_msg[NL802154_ATTR_SDU_LENGTH]) {
+		sdu_len = nla_get_u8(tb_msg[NL802154_ATTR_SDU_LENGTH]);
+		if( sdu_len > BEACON_SDU_LEN_MAX ) {
+			sdu_len = BEACON_SDU_LEN_MAX;
+		}
+		len = sdu_len;
+		printf("SDU Len: %d\n", sdu_len);
+	}
+	else {
+		len = 0;
+	}
+
+	r = parse_nla_array_u8( tb_msg[ NL802154_ATTR_SDU ], NL802154_ATTR_SDU_ENTRY, sdu, &len );
+	if ( 0 != r ) {
+	    goto protocol_error;
+	}
+
+	printf("SDU: ", sdu_len);
+	for (i = 0; i < sdu_len; i++) {
+	    printf("%x ", sdu[i]);
+	}
+	printf("\n", sdu[i]);
+
+	goto out;
+
+protocol_error:
+	fprintf( stderr, "protocol error\n" );
+	r = -EINVAL;
+out:
+	return NL_SKIP;
+}
+
+static int handle_beacon_notify(struct nl802154_state *state,
+				 struct nl_cb *cb,
+				 struct nl_msg *msg,
+				 int argc, char **argv,
+				 enum id_input id)
+{
+    printf("Inside %s %d\n", __FUNCTION__,  atoi(argv[0]) );
+	NLA_PUT_U32(msg, NL802154_ATTR_BEACON_INDICATION_TIMEOUT, atoi(argv[0]));
+	nl_cb_set(cb, NL_CB_VALID, NL_CB_CUSTOM, print_beacon_notify_indication, NULL);
```

Comment:
pretty sure the return value from nl_cb_set is being ignored.

You also need to add scheduled work to timeout and cancel the work. 

Please use existing code from any of the other branches as an example